sql/src/main/java/io/crate/metadata/Schemas.java

```diff
@@ -106,6 +111,25 @@ public DocTableInfo getWritableTable(TableIdent tableIdent) {
             throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
                 "%s is an alias. Write, Drop or Alter operations are not supported", tableInfo.ident()));
         }
+        if (docTableInfo.closed()) {
+            throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
+                "The table %s is closed. Write, Drop or Alter operations (other than OPEN) are not supported",
+                tableInfo.ident()));
+        }
+        return docTableInfo;
+    }
+
+    public DocTableInfo getOpenableOrClosableTable(TableIdent tableIdent) {
+        TableInfo tableInfo = getTableInfo(tableIdent);
+        if (!(tableInfo instanceof DocTableInfo)) {
+            throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
+                "The table %s is read-only. Write, Drop or Alter operations are not supported", tableInfo.ident()));
```

Comment:
@Autophagy I'm not sure but to me it feels as if Schemas is the wrong place for
these kinds of check. That's why I suggested to simplify it on this level. 

I think the table, or a specific schemaInfo instance should have the knowledge
what kind of operations are allowed on them.

The current abstraction/API already doesn't fit without those close changes:
sys.nodes_check can be updated using UPDATE statements, yet the UpdateAnalyzer
can't use getWritableTable because it returns a DocTableInfo.


I think we came up with this method because in some cases we wanted to get a
`DocTableInfo` instance instead of only `TableInfo`.


Also: What happens once we add views or support updates on joined relations?
I think the granularity of operations is more than just write/drop/close. We
probably have to distinguish between data-definition and data-manipulation.

The TableInfo already has a `supportedOperations` set - could we use that?